tests/network/sriov.go

```diff
@@ -575,6 +571,15 @@ var _ = Describe("[Serial]SRIOV", func() {
 	})
 })
 
+func readSRIOVResourceName() string {
+	sriovResourceName := os.Getenv("SRIOV_RESOURCE_NAME")
+	if sriovResourceName == "" {
+		const defaultTestSRIOVResourceName = "kubevirt.io/sriov_net"
```

Comment:
can you explain why do you define the const in this scope, and not outside?

Reply:
I follow the recommendation to define vars/consts as close as possible to the usage.
As a side effect, you now know that no one else is using this const.

Reply:
then why not just use:
```
sriovResourceName = "kubevirt.io/sriov_net"
```
Why do we have to go through const? seems odd

Reply:
Let me rephrase
we use const so that no one could change the var in the future..
but here you create a const and immediately assign the value to a (non-const) variable... 
if the const has no further use and the variable using it is [mutable](https://go.dev/play/p/GqPWnCbDH5k), then why use the const in the first place?

Reply:
> Let me rephrase we use const so that no one could change the const in the future..

A const is processed at compilation and as such has several benefits, one of which is that it is immutable.
But I think one of its other usages is to name scalars and strings, giving them meaning beyond their value.

> but here you create a const and immediately assign the value to a (non-const) variable... if the const has no further use and the variable using it is [mutable](https://go.dev/play/p/GqPWnCbDH5k), then why use the const in the first place?

Because we are interested in explaining what is this `"kubevirt.io/sriov_net"` representing.
We can set in the variable mentioned either this default or a value from an env variable.

Reply:
if they have meaning beyond their value, then shouldn't it live in the broader scope of the test?

Looking at the test code, I don't think we ever do a `os.Setenv("SRIOV_RESOURCE_NAME", value)` so aren't we always getting into this if condition?

Anyway I don't think defining a const in this example and scope is contributing anything, I would add the const in the broader scope and use it here (and perhaps later as well).

Reply:
> if they have meaning beyond their value, then shouldn't it live in the broader scope of the test?

If no one else uses it directly, why would we be interested in increasing its scope?
It expresses the reality, nothing more.
I would expect no one to need it at the moment, just to consume this helper.
If that changes in the future, then the scope needs to change accordingly.

> Looking at the test code, I don't think we ever do a os.Setenv("SRIOV_RESOURCE_NAME", value) so aren't we always getting into this if condition?

I think the idea is for the one who is executing the tests, to control this per the setup they have.

> Anyway I don't think defining a const in this example and scope is contributing anything, I would add the const in the broader scope and use it here (and perhaps later as well).

I answered on why the scope IMO is fine.
And I think it helps readers (including me) to express what this string is.